### DISCUSSION TITLE
Self-hosted runner display (Macos)

### DISCUSSION THREAD

**Discussion by c3-kaspesi** [actions/virtual-environments#5730] [General questions]
I am currently trying to set up a self hosted github action runner since the github hosted runners do not have sufficient memory.  What I've found is the github hosted MacOS runners think they have a display.  Running `displayplacer list` shows... 

```
Persistent screen id: FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF Contextual screen id: 1535231424 Type: 20 inch external screen Resolution: 1176x885 Hertz: N/A Color Depth: 4 Scaling:off Origin: (0,0) - main display Rotation: 0 Resolutions for rotation 0: mode 0: res:1176x885 color_depth:4 <-- current mode Execute the command below to set your screens to the current arrangement: displayplacer "id:FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF res:1176x885 color_depth:4 scaling:off origin:(0,0) degree:0"
```

On my self-hosted mac os runner (running headlessly) I can only get a display to be recognized by setting up vnc and connecting to the machine remotely with screen share.  When no display is recognized running tests on vscode extensions (using electron-test) fails since the machine is running headless.  

How does the github hosted runner emulate a display?  Do the machines have displays / edid emulators?  Do they have a separate machine setting up a vnc connection to create the display?

I need to do this to set up a CI pipeline on the self-hosted github runner, but currently have been unable to find a solution (dont have access to the physical machiens so a edid emulator would not work.  Any help or suggestions on a solution would be appreciated.

**Comment by miketimofeev** [comment]
@c3-kaspesi gh hosted runners use VmWare Vsphere virtualization, no physical displays attached but the display is emulated.

**Reply by c3-kaspesi** [reply]
Thanks!
